core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -4809,6 +4874,30 @@ private static class RegisterArgs {
     }
   }
 
+  /** Determine presence of {@link SqlKind.DEFAULT} type. */
+  private static class DefaultChecker extends SqlShuttle {
```

Comment:
move this into SqlUtil. you should be able to use a pattern very similar to SqlUtil.containsIn